feat(ifr): support pixtral (#3726)
scaleway/docs-content: ai-data/managed-inference/reference-content/pixtral-12b-2409.mdx

  paragraph: This page provides information on the Pixtral-12b-2409 model
tags: 
dates:
  validation: 2024-09-23
categories:
  - ai-data
---

## Model overview

| Attribute       | Details                            |
|-----------------|------------------------------------|
| Provider        | [Mistral](https://mistral.ai/technology/#models)                         |
| Model Name      | `pixtral-12b-2409`       |
| Compatible Instances | H100, H100-2 (bf16)                 |
| Context size | TBDk tokens    |

## Model name

```bash
pixtral-12b-2409:bf16
```

## Compatible Instances

- [H100-1 (bf16)](https://www.scaleway.com/en/h100-pcie-try-it-now/)
- [H100-2 (bf16)](https://www.scaleway.com/en/h100-pcie-try-it-now/)

## Model introduction

Pixtral is a vision language model introducing a novel architecture: 12B parameter multimodal decoder plus 400M parameter vision encoder. 
It can analyze images and offer insights from visual content alongside text. 
This multimodal functionality creates new opportunities for applications that need both visual and textual comprehension. 

Pixtral is open-weight and distributed under the Apache 2.0 license.

## Why is it useful?

> - Pixtral allows to process real world and high resolution images, unlocking capacities such as transcribing handwritten files or payment receipts, extracting information from graphs, captioning images...

Reviewer: ```suggestion
- Pixtral allows you to process real world and high resolution images, unlocking capacities such as transcribing handwritten files or payment receipts, extracting information from graphs, captioning images, etc.
```

> - It offers large context window of up to TBDk tokens, particularly useful for RAG applications. 

Reviewer: ```suggestion
- It offers large context window of up to TBDk tokens, particularly useful for RAG applications
```

--data '{
       "model": "pixtral-12b-2409",
       "messages": [
        {
          "role": "user",
          "content": [
              {"type" : "text", "text": "Describe this image in detail please."},
              {"type": "image_url", "image_url": {"url": "https://picsum.photos/id/32/512/512"}},
              {"type" : "text", "text": "and this one as well. Answer in French."},
              {"type": "image_url", "image_url": {"url": "https://www.wolframcloud.com/obj/resourcesystem/images/a0e/a0ee3983-46c6-4c92-b85d-059044639928/6af8cfb971db031b.png"}}
          ]
        }
       ],
       "top_p": 1, 
       "temperature": 0.7, 
       "stream": false
}'
```

Make sure to replace `<IAM API key>` and `<Deployment UUID>` with your actual [IAM API key](/identity-and-access-management/iam/how-to/create-api-keys/) and the Deployment UUID you are targeting.

<Message type="tip">
  The model name allows Scaleway to put your prompts in the expected format.
</Message>

<Message type="note">
  Ensure that the `messages` array is properly formatted with roles (system, user, assistant) and content.
</Message>

### Passing images to Pixtral

1. Image URLs

> If the image is available online, you can just include the image URL in your request as demonstrated above. This approach is simple and doesn't require any encoding.

Reviewer: ```suggestion
If the image is available online, you can just include the image URL in your request as demonstrated above. This approach is simple and does not require any encoding.
```
